pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -449,7 +439,33 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         return outcome_map;
     }
 
+    void
+    setDeviceSeed(const std::optional<std::size_t> &deviceSeed = std::nullopt) {
+        this->DeviceSeed = deviceSeed;
+    }
+
+    void setDeviceSeedFromPRNG(std::mt19937 *gen) {
+        if (gen != nullptr) {
+            this->setDeviceSeed((*gen)());
+        }
+    }
```

Comment:
Do we really need this method as part of the API? It seems redundant to `setDeviceSeed` as we also need to pass the `gen`. Removing this method in favour of using `setDeviceSeed((*gen)())` directly would avoid a function call and the additional if-condition check. What do you think?

Reply:
Removed